gio/src/subclass/action_group.rs

```diff
@@ -0,0 +1,371 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::ActionGroup;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::{Cast, GString, ObjectExt, Quark, Variant, VariantType};
+use once_cell::sync::Lazy;
+use std::ptr;
+
+pub trait ActionGroupImpl: ObjectImpl {
+    fn action_added(&self, action_group: &Self::Type, action_name: &str);
+    fn action_enabled_changed(&self, action_group: &Self::Type, action_name: &str, enabled: bool);
+    fn action_removed(&self, action_group: &Self::Type, action_name: &str);
+    fn action_state_changed(&self, action_group: &Self::Type, action_name: &str, state: &Variant);
+    fn activate_action(
+        &self,
+        action_group: &Self::Type,
+        action_name: &str,
+        parameter: Option<&Variant>,
+    );
+    fn change_action_state(&self, action_group: &Self::Type, action_name: &str, value: &Variant);
+    fn get_action_enabled(&self, action_group: &Self::Type, action_name: &str) -> bool;
+    fn get_action_parameter_type(
+        &self,
+        action_group: &Self::Type,
+        action_name: &str,
+    ) -> Option<VariantType>;
+    fn get_action_state(&self, action_group: &Self::Type, action_name: &str) -> Option<Variant>;
+    fn get_action_state_hint(
+        &self,
+        action_group: &Self::Type,
+        action_name: &str,
+    ) -> Option<Variant>;
+    fn get_action_state_type(
+        &self,
+        action_group: &Self::Type,
+        action_name: &str,
+    ) -> Option<VariantType>;
+    fn has_action(&self, action_group: &Self::Type, action_name: &str) -> bool;
+    fn list_actions(&self, action_group: &Self::Type) -> Vec<String>;
+    fn query_action(
+        &self,
+        action_group: &Self::Type,
+        action_name: &str,
+    ) -> Option<(bool, VariantType, VariantType, Variant, Variant)>;
```

Comment:
This has missing annotations in C. The 4 last parts of the tuple are all `nullable`.

Reply:
You should probably set them all to `NULL` before returning to C if they were `None` in Rust